src/App.tsx

```diff
@@ -1,31 +1,54 @@
-import useViewStore from './stores/view';
+import useViewStore, { getNavigationProps } from './stores/view';
 import Landing from './components/Landing';
 import Upload from './components/Upload';
 import ColumnAnnotation from './components/ColumnAnnotation';
 import ValueAnnotation from './components/ValueAnnotation';
 import Download from './components/Download';
+import NavigationButton from './components/NavigationButton';
+import StepperComponent from './components/Stepper';
+import { View } from './utils/types';
 
 function App() {
   const currentView = useViewStore((state) => state.currentView);
 
+  const { backView, nextView, backLabel, nextLabel, className } = getNavigationProps(currentView);
+
   const renderView = () => {
     switch (currentView) {
-      case 'landing':
+      case View.Landing:
         return <Landing />;
-      case 'upload':
+      case View.Upload:
         return <Upload />;
-      case 'columnAnnotation':
+      case View.ColumnAnnotation:
         return <ColumnAnnotation />;
-      case 'valueAnnotation':
+      case View.ValueAnnotation:
         return <ValueAnnotation />;
-      case 'download':
+      case View.Download:
         return <Download />;
       default:
         return <Landing />;
     }
   };
 
-  return <div>{renderView()}</div>;
+  return (
+    <div className="flex min-h-screen flex-col overflow-x-hidden">
+      {currentView !== View.Landing && <StepperComponent currentView={currentView} />}
```

Comment:
I'd say let's add a comment here to highlight that this is the top-level nav section / component.

Reply:
Are you sure the comments for these sections necessary? The component/variable names like `NavStepper` and `NavButton` already provide enough clarity don you think?

Reply:
Yeah, I think with the renamed components it'll be clear enough. But since this is more of a "page" component, adding a bit structure in this section doesn't hurt. Up to you, I agree renamed components are probably enough now